File: Medium/1344_Angle_Between_Hands_of_a_Clock/Solution.java



public class Solution {
    public double angleClock(int hour, int minutes) {
        double minutesAng = minutes * 6.0;
        double hourAng = (hour * 30.0) + (minutes * 0.5);
        double diff = Math.abs(minutesAng - hourAng);
        return Math.min(diff, 360.0 - diff);
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        System.out.println(solution.angleClock(12, 30)); // Expected: 165.0
        
        // test cases 2
        System.out.println(solution.angleClock(3, 30)); // Expected: 75.0

        // test cases 3
        System.out.println(solution.angleClock(3, 15)); // Expected: 7.5
        
    }
}

File: Medium/1344_Angle_Between_Hands_of_a_Clock/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    double angleClock(int hour, int minutes) {
        double minutesAng = minutes * 6.0;
        double hourAng = (hour * 30.0) + (minutes * 0.5);
        double diff = abs(minutesAng - hourAng);
        return min(diff, 360.0 - diff);
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    int hour1 = 12, minutes1 = 30;
    cout << solution.angleClock(hour1, minutes1) << endl;
    
    // test cases 2
    int hour2 = 3, minutes2 = 30;
    cout << solution.angleClock(hour2, minutes2) << endl;
    
    // test cases 3
    int hour3 = 3, minutes3 = 15;
    cout << solution.angleClock(hour3, minutes3) << endl;

    return 0;
}